src/main/scala/com/nec/native/CppTranspiler.scala

```diff
@@ -250,29 +252,69 @@ object CppTranspiler {
     )
   }
 
+  private def filterOperation(block: Block): (CodeLines, String) = {
+    /*
+      The last line of the code block is the only line that is transpiled
+      differently because its return value(s) need to be written into the output
+      vecs.
+    */
+    val lastline = block.children.last match {
+      case apply: Apply =>
+        evalApply(apply)
+
+      case select: Select =>
+        evalSelect(select)
+
+      case ifblock: If =>
+        evalIf(ifblock)
+    }
+
+    (
+      CodeLines.from(
+        // Only the first N-1 lines of the block are straightforwardly transpiled
+        block.children.dropRight(1)
+          .map(evalQual)
+          .filter(_.nonEmpty)
+          .map(l => s"${l};"),
+      ),
+      lastline
+    )
+  }
+
   def evalFilterFunc(func: Function): String = {
     val signature = func.veInOutSignature
     val codelines = func.body match {
-      case ident @ Ident(name) =>
-        throw new IllegalArgumentException("Identity filter is not supported (no Boolean vector support)")
+      case ident: Ident =>
+        CodeLines.from(s"${signature.outputs.head.name}[0] = ${evalIdent(ident).replace("_val", "")}[0]->clone();")
```

Comment:
I'm not sure about this case: When I marked it as unsupported, it was mostly because an identity style filter doesn't really make sense to me even if we had boolean vector support. 

I can see where could kinda make sense in a C-style comparison situation (any number != 0 is considered to be "true"), but then it wouldn't compile on the JVM side, because filter only takes `T => Boolean`. 

Letting this resolve to a clone, however doesn't make sense in any case. A clone in filter only ever makes sense on a literal true (as we have in a later case here). 

Am I missing anything?

Reply:
You are correct in that the usefulness of handling the Ident case is very limited - because `evalFilterFunc` is limited to `T => Boolean` expressions, this case will effectively only ever show up in the case of transpiling the identify function for bools `{ (x: Boolean) => x }` 

(Comparisons of `number != 0` would be interpreted as a `Select` or `Apply` case), and a literal `true` or `false` value, e.g. `{ (x: T) => true }` would be handled by the `Literal(Constant(true))` and `Literal(Constant(false))` cases.)

However, I think that feature consistency (i.e. avoid throwing an exception on an otherwise well-formed and handle-able Scala expression) is important to the user experience wherever possible, which is why I implemented this case.

Reply:
> Letting this resolve to a clone, however doesn't make sense in any case. 

>  this case will effectively only ever show up in the case of transpiling the identify function for bools { (x: Boolean) => x }

As this is the filter case, I think the identity function still needs to end up filtering the input, i.e. all false values need to be filtered. A direct `clone` doesn't make sense in this case.

Reply:
Ah, I see what you mean.  Let me update this.

Reply:
This is now fixed.